src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -1324,18 +1324,131 @@ public static unsafe int SequenceCompareTo(ref byte first, int firstLength, ref
             IntPtr offset = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr nLength = (IntPtr)(void*)minLength;
 
-            if (Vector.IsHardwareAccelerated && (byte*)nLength > (byte*)Vector<byte>.Count)
+            if (Avx2.IsSupported)
             {
-                nLength -= Vector<byte>.Count;
-                while ((byte*)nLength > (byte*)offset)
+                if ((byte*)nLength >= (byte*)Vector256<byte>.Count)
                 {
-                    if (LoadVector(ref first, offset) != LoadVector(ref second, offset))
+                    nLength -= Vector256<byte>.Count;
+                    int matches;
+                    while ((byte*)nLength > (byte*)offset)
                     {
-                        goto NotEqual;
+                        matches = Avx2.MoveMask(Avx2.CompareEqual(LoadVector256(ref first, offset), LoadVector256(ref second, offset)));
+                        if (matches == -1)
+                        {
+                            // All matched
+                            offset += Vector256<byte>.Count;
+                            continue;
+                        }
+
+                        goto Difference;
                     }
-                    offset += Vector<byte>.Count;
+                    // Move to Vector length from end for final compare
+                    offset = nLength;
+                    matches = Avx2.MoveMask(Avx2.CompareEqual(LoadVector256(ref first, offset), LoadVector256(ref second, offset)));
+                    if (matches == -1)
+                    {
+                        // All matched
+                        goto Equal;
+                    }
+                Difference:
+                    // Invert matches to find differences
+                    int differences = ~matches;
+                    offset = (IntPtr)((int)(byte*)offset + BitOps.TrailingZeroCount(differences));
```

Comment:
Perhaps I'm misunderstanding, but this seems to be adding a bit offset to a byte offset.

Reply:
Yes `MoveMask`/`vpmovmskb` has converted the equal vector elements into a set of bit flags; so its finding the first flag set (or rather the inverse, then first flag not set to find the first difference) - then adding that to the offset which is the start of the vector data to get the element position.

Reply:
Ah yes, I see. Perhaps a comment, e.g.
```
// Invert matches to find differences.
// Note that MoveMask has converted the equal vector elements into a set of bit flags,
// So the bit position in 'matches' corresponds to the element offset.
```
Or something to that effect to help those of us who might be confused by this.

Reply:
Added; I'll probably be confused by it in a month after not looking :)

Also added to some other places and changed some equality tests to use more the more obvious consts `uint.MaxValue` and `ushort.MaxValue` rather than `-1` and `0xFFFF`

Reply:
Can do better, 1 sec

Reply:
Should be better now?